backend/schedulerd/executor.go

```diff
@@ -327,17 +326,17 @@ func buildRequest(check *types.CheckConfig, s store.Store) (*types.CheckRequest,
 
 	ctx := types.SetContextFromResource(context.Background(), check)
 
+	assets, err := s.GetAssets(ctx, &store.SelectionPredicate{})
+	if err != nil {
+		return nil, err
+	}
+
 	// Guard against iterating over assets if there are no assets associated with
 	// the check in the first place.
 	if len(check.RuntimeAssets) != 0 {
-		// Explode assets; get assets & filter out those that are irrelevant
-		assets, err := s.GetAssets(ctx, &store.SelectionPredicate{})
-		if err != nil {
-			return nil, err
-		}
```

Comment:
I'm not 100% sure why this is moved outside of the `if len(check.RuntimeAssets) != 0 { }` branch. I'm also not 100% sure what that branch is even for :)

Reply:
This call `s.GetAssets(ctx, &store.SelectionPredicate{})` retrieves _all_ assets within the namespace. I moved it outside of the branch to avoid calling the function twice (for checks and for hooks). The `if len(check.RuntimeAssets) != 0` and `if len(hook.RuntimeAssets) != 0` blocks simply verify if the check or hook have any runtime assets associated with their configs. If not, we don't have to iterate through the list to load in the assets.

Reply:
> This call `s.GetAssets(ctx, &store.SelectionPredicate{})` retrieves _all_ assets within the namespace. I moved it outside of the branch to avoid calling the function twice (for checks and for hooks). 

Thanks, that makes perfect sense!

> The `if len(check.RuntimeAssets) != 0` and `if len(hook.RuntimeAssets) != 0` blocks simply verify if the check or hook have any runtime assets associated with their configs. If not, we don't have to iterate through the list to load in the assets.

That's my bad, I read the code too quickly; I now understand why we short-circuit... we're not ranging over the assets/hooks for the current check. We're ranging over _all_ the assets/hooks in the namespace! So the more assets/hooks there are in the namespace, the longer it takes to run through `{asset,hook}IsRelevant()`. I doubt this is a bottleneck for most setup, but ideally I think we should just fetch the needed assets from the store directly, rather than fetch everything and then compare one by one. This is of course completely out of the scope of this PR!

Reply:
I totally agree, there is definitely some room for optimization here. However, at least in the case of hooks, we won't know which assets are actually needed until the check executes (check status influences the hooks which run, and the assets required to run the hook).